lms/static/js/learner_dashboard/views/unenroll_view.js

```diff
@@ -0,0 +1,117 @@
+                         survey = document.querySelector('.options');
+                         for (i = survey.children.length; i >= 0; i--) {
+                             survey.appendChild(survey.children[Math.random() * i | 0]);
+                         }
+                     });
+
+                     $('#unenroll_form').on('ajax:complete', function(event, xhr) {
+                         var errorText,
+                             reasonsSurvey = HtmlUtils.HTML($('.reasons_survey'));
+                         if (xhr.status === 200) {
+                             $('.inner-wrapper header').hide();
+                             $('#unenroll_form').after(HtmlUtils.ensureHtml(reasonsSurvey).toString());
+                             $('.reasons_survey .slide1').removeClass('hidden');
+                             $('#unenroll_form').hide();
+                             $('.submit_reasons').click(function() {
+                                 var reason = $(".reasons_survey input[name='reason']:checked").attr('val');
+                                 if (reason === 'Other') {
+                                     reason = $('.other_text').val();
+                                 }
+                                 if (reason) {
+                                     window.analytics.track('unenrollment_reason.selected', {
+                                         category: 'user-engagement',
+                                         label: reason,
+                                         displayName: 'v1'
+                                     });
+                                 }
+                                 HtmlUtils.setHtml($('.reasons_survey'), HtmlUtils.HTML($('.slide2').html()));
+                                 $('.reasons_survey .return_to_dashboard').attr('href', this.urls.dashboard);
+                                 $('.reasons_survey .browse_courses').attr('href', this.urls.browseCourses);
+                             }.bind(this));
+                         } else if (xhr.status === 403) {
+                             location.href = this.urls.signInUser + '?course_id=' +
+                            encodeURIComponent($('#unenroll_course_id').val()) + '&enrollment_action=unenroll';
+                         } else {
+                             errorText = gettext('An error occurred. Please try again later.');
+                             $('#unenroll_error').html(xhr.responseText ? xhr.responseText : errorText)
+                             .stop()
+                             .css('display', 'block');
+                         }
+                     }.bind(this));
+
+                     $('.action-unenroll').each(function(index) {
+                         var trigger;
+                         $(this).attr('id', 'unenroll-' + index);
+                         // a bit of a hack, but gets the unique selector for the modal trigger
+                         trigger = '#' + $(this).attr('id');
```

Comment:
You pass a string to the DOM and then scrape it from the DOM in the next line. It would be cleaner to assign the id to a variable, add it to the DOM, then use the variable to define the trigger.